AccountTests/FxAClient10Tests.swift

```diff
@@ -151,4 +151,21 @@ class FxAClient10Tests: LiveAccountTest {
         }
         self.waitForExpectations(timeout: 10, handler: nil)
     }
+    
+    func testProfileSuccess() {
+        withVerifiedAccount { emailUTF8, quickStretchedPW in
+            let e = self.expectation(description: "")
+            
+            let stageConfiguration = StageFirefoxAccountConfiguration()
+            let client = FxAClient10(authEndpoint: stageConfiguration.authEndpointURL, oauthEndpoint: stageConfiguration.oauthEndpointURL, profileEndpoint: stageConfiguration.profileEndpointURL)
+            client.login(emailUTF8, quickStretchedPW: quickStretchedPW, getKeys: true) >>== { login in
```

Comment:
Tried several different ways to not use expectations, but all tests failed because of unfulfilled expectations. Ex.
```
let response = client.login(…) >>== { login in
    client.getProfile(withSessionToken: login.value.sessionToken…)
}.value.successValue
XCTAssertNotNil(response.uid)
```
Perhaps I am missing something? However, I did cleanup these tests to make them easier to follow.

Reply:
See e.g., https://github.com/mozilla-mobile/firefox-ios/blob/master/ClientTests/ResetTests.swift#L77

Remember that you shouldn't make an expectation if you're not going to fulfill it — if you're getting that error it implies that you haven't cleaned up the old approach.

Reply:
Cleaned up some of the loose expectations and the tests pass correctly when using

```
        withVerifiedAccount { emailUTF8, quickStretchedPW in
// Empty here
        }
```
This was previously failing when using expectations. However, if I use 

```
        withVerifiedAccount { emailUTF8, quickStretchedPW in
            let stageConfiguration = StageFirefoxAccountConfiguration()
            let client = FxAClient10(authEndpoint: stageConfiguration.authEndpointURL, oauthEndpoint: stageConfiguration.oauthEndpointURL, profileEndpoint: stageConfiguration.profileEndpointURL)
            let response = client.login(emailUTF8, quickStretchedPW: quickStretchedPW, getKeys: true) >>== { login in
                client.getProfile(withSessionToken: login.sessionToken as NSData)
            }.value.successValue
            XCTAssertNotNil(response?.uid)
        }
```

I get the following error.

![screen shot 2017-08-28 at 9 24 52 pm](https://user-images.githubusercontent.com/1295288/29800508-77f70c40-8c38-11e7-8ed2-d6e1c2f4c66d.png)

If I remove the `.value.successValue`, the test fails the assert at the bottom. I am using a valid `signedInUser.json` (verified account). The code  appears to matchup with the examples you linked. Is there something very different with how FxA tests work? I also noticed they are skipped in BuddyBuild.

Reply:
I think your problem is this.

`>>==` is an operator.

So is `.`.

You have, effectively, this:

```
X >>== Y.value.successValue
```

You want this:

```
(X >>== Y).value.successValue
```

Reply:
`Y` in this case is the closure (the bit in `{}`). You can see that the compiler error message is basically saying "you can't call `.value` on a closure-that-returns-`Deferred`".